### PR TITLE
Add ppo randomwalks example

### PR DESCRIPTION
**Pull request by reciprocated**
This pr extends randomwalks example for ppo. Also it changes its reward to be measured in proportion against shortest lengths, which doesn't change much empirically since this is a very easy example, but I find it more interpretable then raw negative lengths.

ILQL: https://wandb.ai/sorry/public/runs/1rl3ls2d
PPO: https://wandb.ai/sorry/public/runs/2vviyvwg

### CODE REVIEW

**Comment by dpaleka** [comment]
Would itbe better to host this model [https://huggingface.co/randomwalks](https://huggingface.co/randomwalks) under [https://huggingface.co/CarperAI](https://huggingface.co/CarperAI) or something similar, if the randomwalks repo is the official test example for experimental features as written in [CONTRIBUTING.md](https://github.com/CarperAI/trlx/blob/15f18f8d610f7e79f76d2fb59c2e463fe387202a/CONTRIBUTING.md?plain=1#L52).

**Comment by reciprocated** [comment]
@jon-tow What do you think about Daniel's comment? Also just for the record I changed setup.cfg to let the top import in randomwalks be both visible from the root and when importing from trlx/sweep.py

File: configs/ilql_randomwalks.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+model:
```
Comment (jon-tow) [inline configs/ilql_randomwalks.yml:1]: Nit: What do you think about moving `ilql_randomwalks.yml` and `ppo_randomwalks.yml` to `examples/randomwalks/configs` (which is something done similarly in the [program synthesis example](https://github.com/CarperAI/trlx/tree/main/examples/experiments/grounded_program_synthesis/config))?

File: configs/ppo_randomwalks.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+model:
+  model_path: "randomwalks/1M"
+  tokenizer_path: "randomwalks/1M"
```
Comment (jon-tow) [inline configs/ppo_randomwalks.yml:3]: Update the paths to point to CarperAI HuggingFace model path.

File: configs/ilql_randomwalks.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  model_path: "randomwalks/1M"
+  tokenizer_path: "randomwalks/1M"
```
Comment (jon-tow) [inline configs/ilql_randomwalks.yml:3]: Update the paths to point to CarperAI HuggingFace model path.

File: examples/randomwalks/randomwalks.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import networkx as nx
```
Comment (jon-tow) [inline examples/randomwalks/randomwalks.py:1]: Do we want to add `networkx` as a dependency? I believe this is what you're referring to in [this](https://github.com/CarperAI/trlx/pull/119#issuecomment-1331418733) comment? I don't see it in the `setup.cfg` 😅 - We can also just point users to `pip install networkx` in the `randomwalks/README.md` which might be better. What do you think?

File: setup.cfg [hunk 5]
```diff
@@ -34,6 +34,5 @@ dev =
 
 [options.packages.find]
 exclude =
-    examples*
```
Comment (jon-tow) [inline setup.cfg:37]: Good idea!

**Review by jon-tow — CHANGES_REQUESTED** [review]
Nice job! I ran everything locally and it worked great 👍 I left some nits if you could address them before merging.

**Comment by jon-tow** [comment]
> @jon-tow What do you think about Daniel's comment? Also just for the record I changed setup.cfg to let the top import in randomwalks be both visible from the root and when importing from trlx/sweep.py

@reciprocated I think it might be easier to find under CarperAI but ultimately it's up to you given that it's your model 😄 If you want to transfer; feel free!

**Comment by LouisCastricato** [comment]
Host under Carper

**Comment by reciprocated** [comment]
@jon-tow Moved configs and moved the model under carper. Also I added networkx as dependency, since it's pretty [lightweight](https://github.com/networkx/networkx/blob/d82815dba6c8ddce19cd49f700298dc82a58f066/requirements/default.txt#L1) and follows a distinct naming pattern 🙂 

ppo run from fresh git clone: https://wandb.ai/sorry/trlx/runs/2kh7e8q6

File: configs/ilql_randomwalks.yml (hunk 1 above)
Comment (reciprocated) [reply configs/ilql_randomwalks.yml]: agree

File: configs/ppo_randomwalks.yml (hunk 2 above)
Comment (reciprocated) [reply configs/ppo_randomwalks.yml]: done

File: examples/randomwalks/randomwalks.py (hunk 4 above)
Comment (reciprocated) [reply examples/randomwalks/randomwalks.py:1]: I was referring to `from examples.randomwalks import generate_random_walks`, but I also added networkx to make it frictionless

**Review by jon-tow — APPROVED** [review]
Looks great to me! 👏

**Pull request merged**